lib/chatbot-api/chatbot-s3-buckets/index.ts

```diff
@@ -1,29 +1,48 @@
 import * as cdk from "aws-cdk-lib";
 import { Construct } from "constructs";
 import * as s3 from "aws-cdk-lib/aws-s3";
+import * as kms from "aws-cdk-lib/aws-kms";
 import { NagSuppressions } from "cdk-nag";
 
+export interface ChatBotS3BucketsProps {
+  readonly retainOnDelete?: boolean;
+  readonly kmsKey?: kms.Key;
+}
+
 export class ChatBotS3Buckets extends Construct {
   public readonly filesBucket: s3.Bucket;
   public readonly userFeedbackBucket: s3.Bucket;
 
-  constructor(scope: Construct, id: string) {
+  constructor(scope: Construct, id: string, props: ChatBotS3BucketsProps) {
     super(scope, id);
 
     const logsBucket = new s3.Bucket(this, "LogsBucket", {
       blockPublicAccess: s3.BlockPublicAccess.BLOCK_ALL,
-      removalPolicy: cdk.RemovalPolicy.DESTROY,
-      autoDeleteObjects: true,
+      removalPolicy:
+        props.retainOnDelete === true
+          ? cdk.RemovalPolicy.RETAIN_ON_UPDATE_OR_DELETE
+          : cdk.RemovalPolicy.DESTROY,
+      autoDeleteObjects: props.retainOnDelete !== true,
       enforceSSL: true,
+      encryption: s3.BucketEncryption.S3_MANAGED,
```

Comment:
Why does logs bucket encrypted with S3 managed key and files bucket has options?

Reply:
It is not supported.
https://docs.aws.amazon.com/AmazonS3/latest/userguide/enable-server-access-logging.html

"You can use [default bucket encryption](https://docs.aws.amazon.com/AmazonS3/latest/userguide/default-bucket-encryption.html) on the destination bucket only if you use server-side encryption with Amazon S3 managed keys (SSE-S3), which uses the 256-bit Advanced Encryption Standard (AES-256). Default server-side encryption with AWS Key Management Service (AWS KMS) keys (SSE-KMS) is not supported."